.github/workflows/blocked-prs.yml

```diff
@@ -0,0 +1,245 @@
+          OWNER=$(echo "$REPOSITORY" | cut -d '/' -f 1)
+          REPO=$(echo "$REPOSITORY" | cut -d '/' -f 2)
+          PR_JSON=$(
+            gh api \
+              -H "Accept: application/vnd.github.raw+json" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              "/repos/$OWNER/$REPO/pulls/$PR_NUMBER"
+          )
+          {
+            echo "REPO=$(jq -r '.base.repo.name' <<< "$PR_JSON")"
+            echo "OWNER=$(jq -r '.base.repo.owner.login' <<< "$PR_JSON")"
+            echo "PR_NUMBER=$(jq -r '.number' <<< "$PR_JSON")"
+            echo "JOB_DATA=$(jq -c '
+              {
+                "repo": .base.repo.name,
+                "owner": .base.repo.owner.login,
+                "repoUrl": .base.repo.html_url,
+                "prNumber": .number,
+                "prHeadSha": .head.sha,
+                "prHeadLabel": .head.label,
+                "prBody": .body,
+                "prLabels": (reduce .labels[].name as $l ([]; . + [$l]))
+              }
+            ' <<< "$PR_JSON")"
+          } >> "$GITHUB_ENV"
+
+      - name: Find Blocked/Stacked PRs in body
+        id: pr_ids
+        run: |
+          prs=$(
+            jq -c '
+              .prBody as $body 
+                | (
+                    $body | 
+                      reduce ( 
+                        . | scan("blocked (?:by|on):? #([0-9]+)") 
+                        | map({
+                          "type": "Blocked on",
+                          "number": ( . | tonumber )
+                        })
+                      ) as $i ([]; . + [$i[]])
+                  ) as $bprs
+                | (
+                    $body | 
+                      reduce ( 
+                        . | scan("stacked on:? #([0-9]+)")
+                        | map({
+                          "type": "Stacked on",
+                          "number": ( . | tonumber )
+                        })
+                      ) as $i ([]; . + [$i[]])
+                  ) as $sprs
+                | ($bprs + $sprs) as $prs 
+                | { 
+                    "blocking": $prs,
+                    "numBlocking": ( $prs | length),
+                  }
+              ' <<< "$JOB_DATA"
+          )
+          echo "prs=$prs" >> "$GITHUB_OUTPUT"
+
+      - name: Collect Blocked PR Data
+        id: blocking_data
+        if: fromJSON(steps.pr_ids.outputs.prs).numBlocking > 0
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          BLOCKING_PRS: ${{ steps.pr_ids.outputs.prs }}
+        run: |
+          blocked_pr_data=$(
+            while read -r pr_data ; do
+              gh api \
+                -H "Accept: application/vnd.github+json" \
+                -H "X-GitHub-Api-Version: 2022-11-28" \
+                "/repos/$OWNER/$REPO/pulls/$(jq -r '.number' <<< "$pr_data")" \
+                | jq -c --arg type "$(jq -r '.type' <<< "$pr_data")" \
+                  '
+                    . | {
```

Comment:
~~We could probably better re-use fetching stuff from PRs in a JS action...~~

Reply:
I will dodge writing a js action for as long as possible. preferably forever.